src/delay_wasm.rs

```diff
@@ -0,0 +1,36 @@
+//! A version of `Delay` that works on wasm.
+
+use gloo_timers::future::TimeoutFuture;
+use send_wrapper::SendWrapper;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+/// A version of `Delay` that works on wasm.
+#[derive(Debug)]
+pub struct Delay(SendWrapper<TimeoutFuture>);
+
+impl Delay {
+    /// Creates a new future which will fire at `dur` time into the future.
+    #[inline]
+    pub fn new(dur: Duration) -> Delay {
+        Self(SendWrapper::new(TimeoutFuture::new(dur.as_millis() as u32)))
+    }
+
+    /// Resets the timeout. Panics on wasm.
```

Comment:
This is inside the wasm impl, so I'm assuming it doesn't panic?
```suggestion
    /// Resets the timeout.
```